File: fusion-apps-mono-repo/apps/backend/fusion-cloud-backend/fusion-cloud/internal/model/organization.go
package model

import "time"

// Organization represents the structure of an organization in the system.
type Organization struct {
	ID           string    `json:"id"`
	Name         string    `json:"name"`
	CreatedAt    time.Time `json:"created_at"`
	Lab          bool      `json:"lab"`
	Partner      string    `json:"partner"`
	Domain       string    `json:"domain"`
	MobileDomain string    `json:"mobile_domain"`
	PricingPlan  string    `json:"pricing_plan"`
	Contacts     struct {
		AdminEmail   string `json:"admin_email"`
		AdminName    string `json:"admin_name"`
		FinanceEmail string `json:"finance_email"`
		FinanceName  string `json:"finance_name"`
	} `json:"contacts"`
	Statistics struct {
		Devices         int `json:"devices"`
		Users           int `json:"users"`
		Groups          int `json:"groups"`
		Spaces          int `json:"spaces"`
		OpenTickets     int `json:"open_tickets"`
		OpenIncidents   int `json:"open_incidents"`
		PendingInvoices int `json:"pending_invoices"`
	} `json:"statistics"`
}

type OrganizationRequest struct {
	Name        string `json:"name" validate:"required"`
	Description string `json:"description,omitempty"`
	MetaData    string `json:"metadata,omitempty"` // JSON string for additional project data
}

type OrganizationResponse struct {
	ID          int64  `json:"id"`
	Name        string `json:"name"`
	Description string `json:"description,omitempty"`
	MetaData    string `json:"metadata,omitempty"` // JSON string for additional project data
	CreatedAt   string `json:"created_at"`
	UpdatedAt   string `json:"updated_at"`
}

File: apps/frontend/cloud-services-ui/packages/xyte-api/src/models/organization.ts
export interface Organization {
    id: string;
    name: string;
    created_at: string;
    lab: boolean;
    partner: string;
    domain: string;
    mobile_domain: string;
    pricing_plan: string;
    contacts: {
        admin_email: string;
        admin_name: string;
        finance_email: string;
        finance_name: string;
    };
    statistics: {
        devices: number;
        users: number;
        groups: number;
        spaces: number;
        open_tickets: number;
        open_incidents: number;
        pending_invoices: number;
    };
}
